### PR TITLE
raise exception properly when there are no disks (bsc#1057430)

### PR DESCRIPTION
**Pull request by wfeldt**
Actually the description of #calculate_proposal is wrong. The space
calculator doesn't raise NoDiskSpaceError but Error if things fail.

Anyway, sticking to the docs and raising NoDiskSpaceError if disks are missing.

### CODE REVIEW

File: src/lib/y2storage/guided_proposal.rb [hunk 1]
```diff
@@ -139,11 +139,11 @@ def calculate_proposal
             exception = error
           end
 
-          return true unless exception
+          return unless exception
```
Comment (ancorgs) [inline src/lib/y2storage/guided_proposal.rb]: What was wrong with returning nil?

**Comment by ancorgs** [comment]
In general it looks ok, apart from the rubocop offence, of course.

File: src/lib/y2storage/guided_proposal.rb (hunk 1 above)
Comment (wfeldt) [reply src/lib/y2storage/guided_proposal.rb]: Returning `true` implies that the return value matters. Then you have to check the code what's really expected. That's just annoying.

Comment (wfeldt) [reply src/lib/y2storage/guided_proposal.rb]: I see - rubocop dislikes it. That's just insane.

Comment (ancorgs) [reply src/lib/y2storage/guided_proposal.rb]: And does it really matter? I mean, with the current code it can only return `nil` (or `true` in your version) or raise an exception. So with the current method API, the returned value looks indeed irrelevant to me.

Don't get me wrong, I'm not against returning nil, true or `:I_did_it` :-) I simply don't see why the returned value is important with the current definition in which this is more a "procedure" than a "function", sort to say.

Comment (wfeldt) [reply src/lib/y2storage/guided_proposal.rb]: Exactly. That's why I wanted to get **rid** of `true`.

Comment (ancorgs) [reply src/lib/y2storage/guided_proposal.rb]: Oh, damn. I have been reading it the other way around all the time, like if you were ADDING the true. Sorry. Looks like I can't distinguish red from green (has happened to me twice this month).

File: test/y2storage/guided_proposal_no_disks.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require_relative "spec_helper"
+require "y2storage"
+
+# stop rspec warning
+RSpec::Expectations.configuration.on_potential_false_positives = :nothing
```
Comment (ancorgs) [inline test/y2storage/guided_proposal_no_disks.rb]: Why is this needed?

Moreover, I assume than placing it here actually affects the whole testsuite (or any test run after this). In that case, it's quite hidden here.

Comment (ancorgs) [reply test/y2storage/guided_proposal_no_disks.rb]: Writing something like `to_not raise_error Y2Storage::Error` below looks less aggressive than disabling the warning (potentially) in the whole testsuite.

Comment (wfeldt) [reply test/y2storage/guided_proposal_no_disks.rb]: Hm, I had expected it to affect only the single test file.

Without this rspec will not let you the error type you catch unspecified without a big warning.

Any better idea?

Comment (wfeldt) [reply test/y2storage/guided_proposal_no_disks.rb]: It's a TypeError, actually. And this looked a bit too specific in a test to me.

Comment (ancorgs) [reply test/y2storage/guided_proposal_no_disks.rb]: Well, if you want a completely generic check, I guess you can check for `Exception` (the root class for any exception)
